api/types/saml_idp_service_provider.go

```diff
@@ -26,9 +26,47 @@ import (
 )
 
 const (
-	unspecifiedNameFormat = "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
-	uriNameFormat         = "urn:oasis:names:tc:SAML:2.0:attrname-format:uri"
-	basicNameFormat       = "urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+	// The following formats are all defined in the SAML 2.0 Core OS Standard.
+	// https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf
```

Comment:
Move the first two comment lines here above the const block since they apply to all of the constants.

Reply:
not all the constants below belong to the `SAML 2.0 Core OS Standard`. Some of them are from `SAML authn context standard` and some are from `SAML technical overview` document. I have updated comment so it just targets the name formats following the comment line. 
```
// The following name formats are defined in the SAML 2.0 Core OS Standard -
```
https://github.com/gravitational/teleport/pull/36967/commits/fd255eb78ac3cbf4ed8a3fae086ce243a6cd734f

Reply:
Sure, but the way godoc works, the info about "the following name formats" only applies to the doc for `SAMLURINameFormat`, when it should apply to a group of items.

Also, convention for godoc is that the doc comment starts with the name of the identifier it documents.

I recommend something like this:

```
// The following name formats are defined in the SAML 2.0 Core OS Standard -
const (
    // SAMLURINameFormat blah blah
    SAMLURINameFormat = "..."
)
```

If there are other constants that are not related to the SAML 2.0 Core OS Standard then just move them into a separate const block.

Reply:
Updated const blocks in https://github.com/gravitational/teleport/pull/36967/commits/7bf09338dd425f9fd8b5d916240b8caa0f823ae9